packages/block-library/src/list/index.js

```diff
@@ -41,3 +42,7 @@ export const settings = {
 	save,
 	deprecated,
 };
+
+export const settings = window.__experimentalEnableListBlockV2
```

Comment:
Oh, this is an interesting way to do it! In the gallery block, the checks / swapping out logic happens within the `edit` and `save` functions. E.g. in [edit-wrapper.js](https://github.com/wordpress/gutenberg/blob/8fb36dea544e74afc5662a75aeebed6be1fc6077/packages/block-library/src/gallery/edit-wrapper.js#L20) and within [saveWithInnerBlocks](https://github.com/wordpress/gutenberg/blob/8fb36dea544e74afc5662a75aeebed6be1fc6077/packages/block-library/src/gallery/save.js#L22), so it happens a little later in the process rather than when the list's `index.js` file is evaluated.

Since you're doing it earlier here, is that why the `__experimentalEnableListBlockV2` flag is set on `window` instead of `window.wp`?

I think it probably makes it easier for your exploration to swap out which settings object you're using as you've done here (also means less changes to the existing list block) so I quite like the approach personally, though it might be better to nest the setting within `window.wp` for consistency sake?

I'll just CC: @glendaviesnz since he dealt with more of the pain points in the gallery v2 settings. There's also the option of exposing the setting in Gutenberg settings as with [__unstableGallleryWithImagesBlocks](https://github.com/wordpress/gutenberg/blob/d5915916abc45e6682f4bdb70888aa41e98aa395/packages/block-editor/src/store/defaults.js#L160), but that approach would likely preclude swapping out settings objects and necessitate the switch happening within the edit and save functions, so I think I'd lean toward the approach in this PR, which feels a bit simpler.

Reply:
Yeh, this seems like a simpler approach than the one we took with the Gallery block refactor.

The other main pain point in the Gallery refactor was difficulties with the mobile apps using/relying on flags to dictate which version to use - flipping the flags was easy in the web client, ie. we could go from defaulting from v1 to v2 with a simple toggle of the flag, but in mobile, the flag checks ended up compiled into app versions so you need to account for the expectations of the older versions of the app in the wild. It was critical to have the flags in the mobile app though as the old version of the editor compiled in lost data when they were confronted with a new version of the block with innerBlocks - but it took a lot of coordination with the mobile team to work out how to transition mobile to the new version. This is most likely stuff you are already aware of as has been also discussed on Quote block PR which has a similar issue, and Dennis was looking at finding a workaround to this issue.